### PR TITLE
Fix two bugs and a perforamnce issue.

### PR DESCRIPTION
**Pull request by ILMTitan**
Fix a performance issue with getting the buckets. (Issue #322).
Fix a navigation bug changing directory to an implied folder that ends with "/".
Fix a bug where an exception is thrown if an accessible project was no longer active.

### CODE REVIEW

**Comment by chrsmith** [comment]
A few comments. Please leave more comments since otherwise we won't know why we made these changes in the future.

Also, "Fix a navigation bug changing directory to an implied folder that ends with '/'" sounds like a great addition. I'd love to see a new integration test that confirms the new behavior works as intended...

**Comment by chrsmith** [comment]
New tests look good. Once the other feedback has been addressed, LGTM.

File: Google.PowerShell/Provider/GoogleCloudStorageProvider.cs [hunk 1]
```diff
@@ -231,7 +229,9 @@ public string RelativePathToChild(string childObjectPath)
         /// <summary>
         /// The Google Cloud Storage service.
         /// </summary>
-        private static StorageService Service { get; } = new StorageService(GCloudCmdlet.GetBaseClientServiceInitializer());
+        private static StorageService Service { get; } = NewService;
+
+        private static StorageService NewService => new StorageService(GCloudCmdlet.GetBaseClientServiceInitializer());
```
Comment (ILMTitan) [inline Google.PowerShell/Provider/GoogleCloudStorageProvider.cs]: Done.

File: Google.PowerShell/Provider/GoogleCloudStorageProvider.cs [hunk 2]
```diff
@@ -254,27 +254,32 @@ public string RelativePathToChild(string childObjectPath)
         /// <summary>
         /// Reports on the usage of the provider.
         /// </summary>
-        private IReportCmdletResults _telemetryReporter;
+        private IReportCmdletResults TelemetryReporter => _telemetryReporter.Value;
```
Comment (ILMTitan) [inline Google.PowerShell/Provider/GoogleCloudStorageProvider.cs]: The provider gets created a lot, and avoiding the disk read the reporter requires seems a good idea. I made it static, so it is only ever created once.

File: Google.PowerShell/Provider/GoogleCloudStorageProvider.cs [hunk 3]
```diff
@@ -919,7 +924,7 @@ private IEnumerable<Object> ListChildren(GcsPath gcsPath, bool recurse, bool all
 
         private static async Task<IEnumerable<Bucket>> ListBucketsAsync(Project project)
         {
-            BucketsResource.ListRequest request = Service.Buckets.List(project.ProjectId);
+            BucketsResource.ListRequest request = NewService.Buckets.List(project.ProjectId);
```
Comment (ILMTitan) [inline Google.PowerShell/Provider/GoogleCloudStorageProvider.cs]: Added comment.

File: Google.PowerShell/Provider/GoogleCloudStorageProvider.cs [hunk 4]
```diff
@@ -942,7 +947,10 @@ private static IEnumerable<Project> ListAllProjects()
                 ListProjectsResponse projects = request.Execute();
                 foreach (Project project in projects.Projects ?? Enumerable.Empty<Project>())
                 {
-                    yield return project;
+                    if (project.LifecycleState == "ACTIVE")
```
Comment (ILMTitan) [inline Google.PowerShell/Provider/GoogleCloudStorageProvider.cs]: Done

File: Google.PowerShell/Provider/GoogleCloudStorageProvider.cs [hunk 5]
```diff
@@ -956,8 +964,8 @@ private IEnumerable<Bucket> ListAllBuckets()
         private static Dictionary<string, Bucket> UpdateBucketCache()
         {
             List<Project> projects = ListAllProjects().ToList();
-            IEnumerable<Bucket> buckets =
-                projects.Select(ListBucketsAsync).ToList().SelectMany(task => task.Result);
+            var tasks = projects.Select(ListBucketsAsync).ToList();
```
Comment (ILMTitan) [inline Google.PowerShell/Provider/GoogleCloudStorageProvider.cs]: Done

**Pull request merged**